### PR TITLE
Add debug info to e2e

### PR DESCRIPTION
**Pull request by petrkotas**
To simplify debuging the e2e pipeline
add the -x option.

Care is taken around the secrets, to print
only required info.

FIXES: #619 

/cc @mjudeikis 

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### CODE REVIEW

**Comment by jim-minter** [comment]
@petrkotas, is this backwards?  I feel like we'd be better trying to get to `set -x` everywhere and then manage the points where we handle secrets and surround them with `set +x` only.  Our CI scripts are already tough and I fear that adding `set -x` in parts only adds more cognitive overhead?

**Comment by mjudeikis** [comment]
Agreed. Set -x by default and cover secrets with 
```
#!/bin/bash -ex
set +x
secret command
set -x
```

File: hack/e2e/run-rp-and-e2e.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash -e
 ######## Helper file to run E2e either locally or using Azure DevOps Pipelines ########
 
+set -x
```
Comment (mjudeikis) [inline hack/e2e/run-rp-and-e2e.sh:4]: why not and line 1?

File: hack/e2e/run-rp-and-e2e.sh [hunk 2]
```diff
@@ -102,6 +105,18 @@ run_e2e() {
     CLUSTER_SPN_SECRET=$(cat $CLUSTERSPN | jq -r .password)
 
     echo "########## 🚀 Create ARO Cluster $CLUSTER - Using client-id : $CLUSTER_SPN_ID ##########"
+    set +x
```
Comment (ehashman) [inline hack/e2e/run-rp-and-e2e.sh:108]: I think this needs to be moved up, with `set -x` even setting the secrets will print:
```
ehashman@red-dot:~$ set -x
ehashman@red-dot:~$ SECRET="xyz"
+ SECRET=xyz
```

File: hack/e2e/run-rp-and-e2e.sh (hunk 1 above)
Comment (jim-minter) [reply hack/e2e/run-rp-and-e2e.sh:4]: because the file is sourced - the shebang is irrelevant in this file.

File: hack/e2e/run-rp-and-e2e.sh [hunk 3]
```diff
@@ -102,6 +105,18 @@ run_e2e() {
...
+    echo "az aro create" \
+        "-g $ARO_RESOURCEGROUP" \
+        "-n $CLUSTER" \
+        "-l $LOCATION" \
+        "--vnet dev-vnet" \
+        "--master-subnet $CLUSTER-master" \
+        "--worker-subnet $CLUSTER-worker" \
+        "--client-id CLUSTER_SPN_ID" \
+        "--client-secret CLUSTER_SPN_SECRET" \
+        "--cluster-resource-group $CLUSTER_RESOURCEGROUP"
```
Comment (jim-minter) [inline hack/e2e/run-rp-and-e2e.sh:118]: Madness lies in this direction.  Let's not do this.

File: hack/e2e/run-rp-and-e2e.sh (hunk 1 above)
Comment (jim-minter) [inline hack/e2e/run-rp-and-e2e.sh:4]: I'm not wild about this side-effect: any caller who sources this file and then handles secrets afterwards will leak their secret.

The PR should have all callers do `set -x` explicitly.

**Pull request merged**